ui/uihandler.go

```diff
@@ -187,95 +189,110 @@ func Handler(w http.ResponseWriter, r *http.Request) {
 	}
 	o := fhttp.HTTPRunnerOptions{
 		RunnerOptions: ro,
-		URL:           url,
+		HTTPOptions:   *opts,
 	}
 	res, err := fhttp.RunHTTPTest(&o)
 	if err != nil {
-		w.Write([]byte(fmt.Sprintf("Aborting because %v\n", err))) // nolint: errcheck
+		w.Write([]byte(fmt.Sprintf("Aborting because %v\n", err))) // nolint: errcheck,gas
 		return
 	}
+	json, err := json.MarshalIndent(res, "", "  ")
+	if err != nil {
+		log.Fatalf("Unable to json serialize result: %v", err)
+	}
 	if JSONOnly {
 		w.Header().Set("Content-Type", "application/json")
-		j, err := json.MarshalIndent(res, "", "  ")
-		if err != nil {
-			log.Fatalf("Unable to json serialize result: %v", err)
-		}
-		_, err = w.Write(j)
+		_, err = w.Write(json)
 		if err != nil {
 			log.Errf("Unable to write json output for %v: %v", r.RemoteAddr, err)
 		}
 		return
 	}
-	// nolint: errcheck
+	if DoSave {
+		res := SaveJSON(formatDate(&res.StartTime), json)
+		if res != "" {
+			// nolint: errcheck, gas
+			w.Write([]byte(fmt.Sprintf("Saved result to <a href='%s'>%s</a>\n", res, res)))
+		}
+	}
+	// nolint: errcheck, gas
 	w.Write([]byte(fmt.Sprintf("All done %d calls %.3f ms avg, %.1f qps\n</pre>\n<script>\n",
 		res.DurationHistogram.Count,
 		1000.*res.DurationHistogram.Avg,
 		res.ActualQPS)))
-	ResultToJsData(w, res)
-	ResultToChart(w, res)
-	w.Write([]byte("</script></body></html>\n"))
+	ResultToJsData(w, json)
+	w.Write([]byte("</script></body></html>\n")) // nolint: gas
 }
 
-// ResultToJsData converts a result object to chart data arrays.
-func ResultToJsData(w io.Writer, res *fhttp.HTTPRunnerResults) {
-	w.Write([]byte(`var dataP = [{x: 0.0, y: 0.0}, `)) // nolint: errcheck
-	for i, it := range res.DurationHistogram.Data {
-		var x float64
-		if i == 0 {
-			// Extra point, 1/N at min itself
-			x = 1000. * it.Start
-			// nolint: errcheck
-			w.Write([]byte(fmt.Sprintf("{x: %.12g, y: %.3f},\n", x, 100./float64(res.DurationHistogram.Count))))
-		}
-		if i == len(res.DurationHistogram.Data)-1 {
-			//last point we use the end part (max)
-			x = 1000. * it.End
-		} else {
-			x = 1000. * (it.Start + it.End) / 2.
-		}
-		// nolint: errcheck
-		w.Write([]byte(fmt.Sprintf("{x: %.12g, y: %.3f},\n", x, it.Percent)))
+// ResultToJsData converts a result object to chart data arrays and title
+// and creates a chart from the result object
+func ResultToJsData(w io.Writer, json []byte) {
+	// nolint: errcheck, gas
+	w.Write([]byte("var res = "))
+	// nolint: errcheck, gas
+	w.Write(json)
```

Comment:
Is JSON always convertible to JS code? Safest should be "JSON.parse('%s')" % json

Reply:
I though json was safe (and it works so far) - I tried to insert JSON.parse(' ... ') but that errors out with
Uncaught SyntaxError: Invalid or unexpected token

will leave as is for now